feat: add json output for `gh pr checks`

nobe4 opened the pull request:
This PR proposes JSON ouptup support for the `gh pr checks` command.

Unlike the suggestion in https://github.com/cli/cli/issues/6056#issuecomment-1218390676, it doesn't do any aggregation, and rather returns the plain values.

The fields were extracted from https://github.com/cli/cli/blob/178fc2e51ead22bd0ef2904b4b5e3018d657914d/pkg/cmd/pr/checks/aggregate.go#L12-L20

The logic was inspired by https://github.com/cli/cli/blob/4b077daf7e74dbe6bf933f69e220066e3a12e8b4/pkg/cmd/repo/deploy-key/list/list.go upon suggestions from @andyfeller.

Fixes https://github.com/cli/cli/issues/6056

nobe4 committed:
feat: add json output for PR checks

nobe4 committed:
Merge branch 'trunk' into pr_checks_json

nobe4 commented on pkg/cmd/pr/checks/checks.go:
```diff
@@ -161,6 +176,10 @@ func checksRun(opts *ChecksOptions) error {
 		return err
 	}
 
+	if opts.Exporter != nil {
+		return opts.Exporter.Write(opts.IO, checks)
+	}
```
This doesn't seem to do any filtering on `--json` flag. E.g.:

```shell
$ go run cmd/gh/main.go pr checks https://github.com/cli/cli/pull/9079 --json name | jq | head -n 15
[
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:02:08Z",
    "completedAt": "2024-05-14T06:02:21Z",
    "link": "https://github.com/cli/cli/actions/runs/9074506128/job/24933385414",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "IN_PROGRESS",
```

However, based on https://github.com/cli/cli/blob/4b077daf7e74dbe6bf933f69e220066e3a12e8b4/pkg/cmd/repo/deploy-key/list/list.go#L75-L77, it seems that no extra code is required to do the work. So I'm confused about what's missing 🤔

andyfeller replied:
```shell
andyfeller@Andys-MBP:cli/cli ‹pr_checks_json›$ make
go build -trimpath -ldflags "-X github.com/cli/cli/v2/internal/build.Date=2024-05-14 -X github.com/cli/cli/v2/internal/build.Version=v2.49.2-3-g8a1995c9 " -o bin/gh ./cmd/gh
go: downloading google.golang.org/protobuf v1.34.1
go: downloading github.com/sigstore/protobuf-specs v0.3.2


andyfeller@Andys-MBP:cli/cli ‹pr_checks_json›$ ./bin/gh pr checks 9079 --json
Specify one or more comma-separated fields for `--json`:
  bucket
  completedAt
  description
  event
  link
  name
  startedAt
  state
  workflow


andyfeller@Andys-MBP:cli/cli ‹pr_checks_json›$ ./bin/gh pr checks 9079 --json name,state
[
  {
    "name": "CodeQL",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:09:38Z",
    "completedAt": "2024-05-14T06:09:42Z",
    "link": "https://github.com/cli/cli/runs/24933625166",
    "bucket": "pass",
    "event": "",
    "workflow": "",
    "description": ""
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:47Z",
    "completedAt": "2024-05-14T06:18:28Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933550081",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:23Z",
    "completedAt": "2024-05-14T06:15:00Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550650",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:23Z",
    "completedAt": "2024-05-14T06:08:52Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550798",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "lint",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:21Z",
    "completedAt": "2024-05-14T06:08:57Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565587/job/24933550288",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Lint",
    "description": ""
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:20Z",
    "completedAt": "2024-05-14T06:11:59Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549727",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "CodeQL-Build",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:20Z",
    "completedAt": "2024-05-14T06:09:47Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565586/job/24933550178",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Code Scanning",
    "description": ""
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:19Z",
    "completedAt": "2024-05-14T06:07:32Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550518",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:19Z",
    "completedAt": "2024-05-14T06:07:43Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933551059",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "lint",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:18Z",
    "completedAt": "2024-05-14T06:08:53Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565383/job/24933549315",
    "bucket": "pass",
    "event": "push",
    "workflow": "Lint",
    "description": ""
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:18Z",
    "completedAt": "2024-05-14T06:10:35Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550239",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:18Z",
    "completedAt": "2024-05-14T06:11:14Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565588/job/24933550925",
    "bucket": "pass",
    "event": "pull_request",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:17Z",
    "completedAt": "2024-05-14T06:10:32Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549552",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:17Z",
    "completedAt": "2024-05-14T06:07:44Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549902",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:16Z",
    "completedAt": "2024-05-14T06:07:29Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933549370",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS",
    "startedAt": "2024-05-14T06:07:16Z",
    "completedAt": "2024-05-14T06:10:44Z",
    "link": "https://github.com/cli/cli/actions/runs/9074565382/job/24933550257",
    "bucket": "pass",
    "event": "push",
    "workflow": "Unit and Integration Tests",
    "description": ""
  }
]
```

Let's take a look as this should only be exporting the fields specified.

andyfeller replied:
After pushing up the `ExportData` implementation, this is looking better:

```shell
andyfeller@Andys-MBP:cli/cli ‹pr_checks_json*›$ ./bin/gh pr checks 9079 --json name,state
[
  {
    "name": "CodeQL",
    "state": "SUCCESS"
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "lint",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (windows-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "CodeQL-Build",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "lint",
    "state": "SUCCESS"
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (macos-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "integration-tests (ubuntu-latest)",
    "state": "SUCCESS"
  },
  {
    "name": "build (macos-latest)",
    "state": "SUCCESS"
  }
]
```

nobe4 replied:
I concur:
```shell
$ go run cmd/gh/main.go pr checks https://github.com/cli/cli/pull/9079 --json name
[
  {
    "name": "CodeQL"
  },
  {
    "name": "build (windows-latest)"
  },
  {
    "name": "build (windows-latest)"
   },
   ...
]
```

nobe4 replied:
This can be closed, thanks @andyfeller for finding the fix!

nobe4 committed:
fix: rename fields list

andyfeller committed:
Implement ExportData to filter json fields

In order to filter json fields, the `ExportData` interface needed to be implemented with logic that iterated on the selected fields.

andyfeller commented on pkg/cmd/pr/checks/aggregate.go:
```diff
@@ -28,6 +29,10 @@ type checkCounts struct {
 	Canceled int
 }
 
+func (ch *check) ExportData(fields []string) map[string]interface{} {
+	return cmdutil.StructExportData(ch, fields)
+}
+
```
**Is there anything about how PR status check roll ups are exported that should be considered here?**

https://github.com/cli/cli/blob/178fc2e51ead22bd0ef2904b4b5e3018d657914d/api/export_pr.go#L48-L76

```shell
$ gh pr view 9079 --json statusCheckRollup
{
  "statusCheckRollup": [
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:06Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558491/job/24952304248",
      "name": "CodeQL-Build",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Code Scanning"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:52:03Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558501/job/24952304198",
      "name": "lint",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Lint"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:52:04Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557717/job/24952302132",
      "name": "lint",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Lint"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:53Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952304215",
      "name": "build (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:53Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302733",
      "name": "build (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:35Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "0001-01-01T00:00:00Z",
      "conclusion": "",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305470",
      "name": "build (windows-latest)",
      "startedAt": "2024-05-14T13:50:40Z",
      "status": "IN_PROGRESS",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "0001-01-01T00:00:00Z",
      "conclusion": "",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303574",
      "name": "build (windows-latest)",
      "startedAt": "2024-05-14T13:50:43Z",
      "status": "IN_PROGRESS",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:26Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306154",
      "name": "build (macos-latest)",
      "startedAt": "2024-05-14T13:50:35Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:20Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304747",
      "name": "build (macos-latest)",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:50:49Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305094",
      "name": "integration-tests (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:50:49Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302126",
      "name": "integration-tests (ubuntu-latest)",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:51Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305791",
      "name": "integration-tests (windows-latest)",
      "startedAt": "2024-05-14T13:50:40Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:54:00Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303150",
      "name": "integration-tests (windows-latest)",
      "startedAt": "2024-05-14T13:50:39Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:51:05Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306752",
      "name": "integration-tests (macos-latest)",
      "startedAt": "2024-05-14T13:50:38Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:51:03Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304015",
      "name": "integration-tests (macos-latest)",
      "startedAt": "2024-05-14T13:50:36Z",
      "status": "COMPLETED",
      "workflowName": "Unit and Integration Tests"
    },
    {
      "__typename": "CheckRun",
      "completedAt": "2024-05-14T13:53:00Z",
      "conclusion": "SUCCESS",
      "detailsUrl": "https://github.com/cli/cli/runs/24952426191",
      "name": "CodeQL",
      "startedAt": "2024-05-14T13:52:57Z",
      "status": "COMPLETED",
      "workflowName": ""
    }
  ]
}
```

versus

```shell
$ ./bin/gh pr checks 9079 --json bucket,completedAt,description,event,link,name,startedAt,state,workflow
[
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:00Z",
    "description": "",
    "event": "",
    "link": "https://github.com/cli/cli/runs/24952426191",
    "name": "CodeQL",
    "startedAt": "2024-05-14T13:52:57Z",
    "state": "SUCCESS",
    "workflow": ""
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:59:59Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303574",
    "name": "build (windows-latest)",
    "startedAt": "2024-05-14T13:50:43Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pending",
    "completedAt": "0001-01-01T00:00:00Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305470",
    "name": "build (windows-latest)",
    "startedAt": "2024-05-14T13:50:40Z",
    "state": "IN_PROGRESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:51Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305791",
    "name": "integration-tests (windows-latest)",
    "startedAt": "2024-05-14T13:50:40Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:00Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952303150",
    "name": "integration-tests (windows-latest)",
    "startedAt": "2024-05-14T13:50:39Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:50:49Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952305094",
    "name": "integration-tests (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:51:05Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306752",
    "name": "integration-tests (macos-latest)",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:53Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952304215",
    "name": "build (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:06Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558491/job/24952304248",
    "name": "CodeQL-Build",
    "startedAt": "2024-05-14T13:50:38Z",
    "state": "SUCCESS",
    "workflow": "Code Scanning"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:20Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304747",
    "name": "build (macos-latest)",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:50:49Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302126",
    "name": "integration-tests (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:52:04Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557717/job/24952302132",
    "name": "lint",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Lint"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:51:03Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952304015",
    "name": "integration-tests (macos-latest)",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:52:03Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558501/job/24952304198",
    "name": "lint",
    "startedAt": "2024-05-14T13:50:36Z",
    "state": "SUCCESS",
    "workflow": "Lint"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:54:26Z",
    "description": "",
    "event": "pull_request",
    "link": "https://github.com/cli/cli/actions/runs/9080558500/job/24952306154",
    "name": "build (macos-latest)",
    "startedAt": "2024-05-14T13:50:35Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  },
  {
    "bucket": "pass",
    "completedAt": "2024-05-14T13:53:53Z",
    "description": "",
    "event": "push",
    "link": "https://github.com/cli/cli/actions/runs/9080557713/job/24952302733",
    "name": "build (ubuntu-latest)",
    "startedAt": "2024-05-14T13:50:35Z",
    "state": "SUCCESS",
    "workflow": "Unit and Integration Tests"
  }
]
```

nobe4 replied:
Unless there's a specific reason, I think having parity between the two outputs would be better. It seems that the `statusCheckRollup` is missing `bucket`, `description`, and `event` and has some different names 🤔 

`gh pr checks --json` | `gh pr view --json statusCheckRollup`
---                   | ---
bucket                |
completedAt           | completedAt
description           |
event                 |
link                  | detailsUrl
name                  | name
startedAt             | startedAt
state                 | conclusion
workflow              |
_                     | status

andyfeller replied:
Agree being consistent is valuable.

nobe4 replied:
How would you feel about creating a new issue for this specifically? I feel that it warrants a longer discussion that would probably benefits from its own issue, and this PR can move forward?

andyfeller replied:
That's fair.

andyfeller replied:
created https://github.com/cli/cli/issues/9091 to follow up

andyfeller reviewed (approved):
Thank you again, @nobe4, for your contribution! ❤️

findepi commented on pkg/cmd/pr/checks/checks.go:
```diff
@@ -20,10 +20,23 @@ import (
+	"startedAt",
+	"completedAt",
+	"link",
+	"bucket",
```
I know this is not the right place to ask this, but can't find the better one and https://cli.github.com/manual/gh_pr_checks is light on details.
What is the `bucket` field? Looking at the values it has, it's similar to status, right?

nobe4 replied:
FWIW, it's defined and filled in https://github.com/cli/cli/blob/e36d8f103d5e1fb3989d9d5aefdae36b01fb88d4/pkg/cmd/pr/checks/aggregate.go#L72-L88

It seems to be grouping checks by results

andyfeller replied:
@findepi : I think this is a great question and might be worth opening a new issue for improving the `--help` output to include a little more information about buckets.

findepi replied:
thanks @nobe4 for the link, the code suggests i want to use this field for filtering
thanks @andyfeller for suggestion -- https://github.com/cli/cli/issues/9205
